expcore/Mini_games.lua

```diff
@@ -199,75 +211,77 @@ function Mini_games.error_in_game(error_game)
     game.print("an error has occured things may be broken, error: "..error_game)
 end
 local mini_game_list
+--gui
 local on_vote_click = function (player,element,event)
-    --local frame = Gui.get_left_element(player,mini_game_list).container
-    --local scroll = frame.scroll.table
-    local caption = tonumber(element.parent.parent["edit-1"]["race"].caption)+1
-    element.parent.parent["edit-1"]["race"].caption = tostring(caption)
+    local name = element.parent.name
+    local scroll_table = element.parent.parent
+    local mini_game = Mini_games.mini_games[name]
+    local args
+    if mini_game.gui_callback then
+        args = mini_game.gui_callback(scroll_table)
+    end
+    --main_gui[1] = element.parent.parent.parent.parent.parent.parent
+    for i,player in ipairs(game.connected_players) do
+        main_gui[i] = Gui.get_left_element(player,mini_game_list)
+        Gui.toggle_left_element(player,main_gui[1],false)
+    end
+    if args then
+        Mini_games.start_game(name,args)
+    else 
+        Mini_games.start_game(name)
+    end
```

Comment:
Are you testing for nil here? If so then you dont need to test since if it is nil then you would be passing nil as the second argument in both caes, making the two calles to start_game identical.

Reply:
yep checking nill ill test if your right about being able to remove it.